augur/filter/_run.py

```diff
@@ -137,7 +137,7 @@ def run(args):
 
     # Setup strain output.
     if args.output_strains:
-        output_strains = open_file(args.output_strains, "w")
+        output_strains = open(args.output_strains, "w", encoding=ENCODING)
```

Comment:
I don't love this because it breaks the consistency of `open_file()`, which begs the question "Why?". It also removes the compression handling from this point and thus makes that spotty in a weird way that's solely dependent on how we wrote some code instead of some functional reason.

What went wrong with explicitly calling `.__enter__()` here? It works if I try it…

Also, we should be able to make `open_file()` work without being a context manager at all if we moved the `UnicodeDecodeError` handling out of it and into a higher scope. (It would still return objects which support the context manager protocol.)

Reply:
Upon reverting 1d00c532bcf3f9f0c2ecac5093dd000ad9769bf6 and applying the patch

```diff
- output_log = open_file(args.output_log, "w", newline='')
+ output_log = open_file(args.output_log, "w", newline='').__enter__()

```

I got an error when running tests/functional/filter/cram/filter-metadata-sequence-strains-mismatch.t.


```
File "…/python3.10/csv.py", line 154, in writerow
  return self.writer.writerow(self._dict_to_list(rowdict))
ValueError: I/O operation on closed file.
```

Turns out that this was because the thing that is returned by `open_file()` (context manager?) was being lost, maybe due to garbage collection since it was used in an `if` block. Fixed by keeping that around in the scope of `run()`.

https://github.com/nextstrain/augur/blob/c2ff05e9acc8426db466cd7010572b4364288e07/augur/filter/_run.py#L137

Note that this no longer applies to `--output-strains` after rebase onto 982bc1c93861d9e6e7b33d9a68e75b540fbc2938, but still applies to `--output-log`.

Reply:
Ahh, yeah, makes sense it would be lost due to automatic reference counting.